Rubberduck.VBEEditor/Utility/SelectionService.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using Rubberduck.VBEditor.ComManagement;
+using Rubberduck.VBEditor.SafeComWrappers.Abstract;
+
+namespace Rubberduck.VBEditor.Utility
+{
+    public class SelectionService : ISelectionService
+    {
+        private readonly IProjectsProvider _projectsProvider;
+        private readonly IVBE _vbe;
+
+        public SelectionService(IVBE vbe, IProjectsProvider projectsProvider)
+        {
+            _vbe = vbe;
+            _projectsProvider = projectsProvider;
+        }
+
+
+        public QualifiedSelection? ActiveSelection()
+        {
+            using (var activeCodePane = _vbe.ActiveCodePane)
+            {
+                return activeCodePane?.GetQualifiedSelection();
+            }
+        }
+
+        public Selection? Selection(QualifiedModuleName module)
+        {
+            var component = _projectsProvider.Component(module);
+            if (component == null)
+            {
+                return null;
+            }
+
+            using (var codeModule = component.CodeModule)
+            using (var codePane = codeModule.CodePane)
+            {
+                return codePane.Selection;
+            }
+        }
+
+        public bool TryActivate(QualifiedModuleName module)
+        {
+            try
+            {
+                var component = _projectsProvider.Component(module);
+                if (component == null)
+                {
+                    return false;
+                }
+
+                using (var codeModule = component.CodeModule)
+                using(var codePane = codeModule.CodePane)
+                {
+                    _vbe.ActiveCodePane = codePane;
+                }
+
+                return true;
+            }
+            catch (Exception)
+            {
+                return false;
```

Comment:
I will add a 'debug` log. (Trace would get drowned.)

I am catching all here primarily because it is unexpected for a `Try*` to throw. However, I am considering to narrow the catch down to `ComException`, which was my initial intention, thoughts?

Reply:
I agree with wrapping a try/catch. 

My only concern is that the `COMException` itself is quite board. It might make more sense to additionally filter on the HRESULT and thus ignore HRESULTs that we know that just means not there and keep a generic catch to write the debug entry for some unexpected errors that shouldn't have come up in the failed attempt to retrieve a component.

Reply:
To clarify, there will never be an exception from retrieving the component. The projects repository should never throw on retrieval, only possibly on refresh.

I am concerned here with whatever wired thing goes wrong when dealing with the COM interop.

Reply:
But wouldn't that fall more into the category of "errors we weren't expecting and probably should know about", at least initially? We can then make decisions to exclude the error once we've investigated and found this to be a normal outcome and update the catch block to ignore that particular error altogether. For that purpose, there might be no promises that it's a `COMException` and as stated before, a `Try*` method should never throw so we still will need a catch-all block and at least log it for further analysis.